modules/programs/git.nix

```
@@ -407,7 +407,10 @@ in {
                 mkIf smtp.tls.enable (toString smtp.tls.certificatesFile);
               smtpServer = smtp.host;
               smtpUser = userName;
-              from = address;
+              from = if (builtins.stringLength realName == 0) then
```

I think it should be safe to simply assume that `realName` is set, it does not have a default value so people would have to manually set `realName = ""` in their configuration for it to be an empty string.

But if you want to keep this if-expression, then you can simplify the condition
```suggestion
              from = if realName == "" then
```

---

Oh yes I think misread the module's doc 😆 

Would you be ok with:
```suggestion
              from = if account ? realName then
```

---

I think the best would be to simply do without the if-expression:

``` nix
from = "${realName} <${address}>";
```

Since the `realName` option is non-nullable it should be always set by the user.